### PR TITLE
Irohad field reorder

### PR DESCRIPTION
**Pull request by luckychess**
### Description of the Change
Field declarations changed according their initialisation order and a bit of code cleaning.

### Benefits
A bit more consistent order of destructor calls.

### Possible Drawbacks 
None.

### CODE REVIEW

File: irohad/main/application.cpp [hunk 1]
```diff
@@ -514,9 +514,3 @@ Irohad::RunResult Irohad::run() {
             return e;
           });
 }
-
-Irohad::~Irohad() {
```
Comment (Akvinikym) [inline irohad/main/application.cpp]: Why is `freeConnections()` function call no longer needed?

Comment (luckychess) [reply irohad/main/application.cpp]: It's called in Storage destructor, no need to do it manually from here.

File: irohad/main/application.hpp [hunk 2]
```diff
@@ -166,6 +166,24 @@ class Irohad {
...
+ protected:
+  logger::Logger log_;
+
+  // initialization objects
+  iroha::network::OnDemandOrderingInit ordering_init;
+  iroha::consensus::yac::YacInit yac_init;
+  iroha::network::BlockLoaderInit loader_init;
+
+  // common objects factory
+  std::shared_ptr<shared_model::interface::CommonObjectsFactory>
```
Comment (lebdron) [inline irohad/main/application.hpp]: Technically `common_objects_factory_` should be above `std::shared_ptr<iroha::ametsuchi::Storage> storage`
https://github.com/hyperledger/iroha/blob/d617982653b0cf8b03e349f937fafac37bce37e2/irohad/main/application.cpp#L122-L136

Comment (luckychess) [reply irohad/main/application.hpp]: I agree but I don't really want to do a protected-public-protected mix of fields. I think it could be done later.

**Pull request merged**